plasmapy/diagnostics/path_integrated_diagnostic.py

```diff
@@ -0,0 +1,412 @@
+            pos_out[0] = x.to(u.m).value
+            pos_out[1] = y.to(u.m).value
+            pos_out[2] = z.to(u.m).value
+
+        elif geometry == "cylindrical":
+            r, t, z = pos
+            r = r.to(u.m)
+            t = t.to(u.rad).value
+            z = z.to(u.m)
+            pos_out[0] = (r * np.cos(t)).to(u.m).value
+            pos_out[1] = (r * np.sin(t)).to(u.m).value
+            pos_out[2] = z.to(u.m).value
+
+        elif geometry == "spherical":
+            r, t, p = pos
+            r = r.to(u.m)
+            t = t.to(u.rad).value
+            p = p.to(u.rad).value
+
+            pos_out[0] = (r * np.sin(t) * np.cos(p)).to(u.m).value
+            pos_out[1] = (r * np.sin(t) * np.sin(p)).to(u.m).value
+            pos_out[2] = (r * np.cos(t)).to(u.m).value
+
+        return pos_out
+
+    def _default_detector_hdir(self):
+        """
+        Calculates the default horizontal unit vector for the detector plane
+        (see __init__ description for details)
+        """
+        # Create unit vectors that define the detector plane
+        # Define plane  horizontal axis
+        if np.allclose(np.abs(self.det_n), np.array([0, 0, 1])):
+            nx = np.array([1, 0, 0])
+        else:
+            nx = np.cross(np.array([0, 0, 1]), self.det_n)
+        nx = nx / np.linalg.norm(nx)
+        return nx
+
+
+class LineIntegratedDiagnostic(PathIntegratedDiagnostic):
+    def _line_integral(
+        self,
+        size=np.array([[-1, 1], [-1, 1]]) * u.cm,
+        bins=[50, 50],
+        collimated=True,
+        num=100,
+    ):
+        """
+        Calculates the line-integral tof the integrand function through the
+        provided grid. This is accomplished through the following steps:
+
+        1) Create an array of points in the detector plane using the detector
+        location and the size and bins keywords.
+
+        2) For each cell of detector grid, create an array of points of
+        separation ds from there to the source point (or, when collimated=True,
+                                                      the source plane).
+
+        3) Evaluate the integrand function at each point.
+
+        4) Integrate along the line-integrated dimension to obtain the
+        line-integrated quantity in the detector plane.
+
+        Parameters
+        ----------
+        size : `~astropy.units.Quantity` array of shape [2,2]
+            The bounds of the detector region. The default is [[-1,1],[-1,1]] cm.
+
+        bins : integer ndarray array of shape [2,2]
+            Number of bins in each direction of the detector region. The
+            default is [50,50].
+
+        collimated : Boolean, optional
+            If True, the source will be assumed to be collimated. If False,
+            a point source will be used. The default is True (collimated).
+
+        num : int, optional
+            Number of integration points along the line (within the grid region).
+            The default is 100.
```

Comment:
Seems like a potentially fragile default...? I imagine if someone has steep gradients, it might cause incorrect results.

Reply:
Maybe it makes sense to just make this a required input? 100 could be overkill for some grids or insufficient for others. Or it could default to `npmax(grid.shape)` or something?